elk/training/reporter.py

```diff
@@ -325,15 +331,17 @@ def train_loop_adam(
             probe.parameters(), lr=cfg.lr, weight_decay=cfg.weight_decay
         )
 
-        loss = torch.inf
+        loss: Optional[Tensor] = None
         for _ in range(cfg.num_epochs):
             optimizer.zero_grad()
 
             loss = self.loss(probe(x0), probe(x1), labels)
             loss.backward()
             optimizer.step()
 
-        return float(loss)
+        return (
+            float(loss) if loss is not None else raise_assertion_error("Loss is None")
+        )
```

Comment:
i'm raising an assertion error here, otherwise we can keep the original implementation of returning torch.inf if num_epochs is somehow 0. I think its better to raise earlier

Reply:
We should probably just raise before the for-loop even starts if `num_epochs < 1`. Would that fix the mypy error?

Reply:
i don't think i can be fixed. because there is no "type-system" thing that says num_epochs will be a positive integer >= 1. so even if you raise before that, you can't prove to the type system that "this will definitely loop one time".

IDK why pyright doesn't complain though.

Reply:
ok actually i'll try it out by reverting to torch.inf